rollup.config.js

```diff
@@ -122,15 +125,85 @@ function resolveGenerated() {
     name: "resolve-msg-generated",
     resolveId(importee) {
       if (importee.startsWith("gen/msg_generated")) {
-        const resolved = path.resolve(
-          path.join(process.cwd(), `${importee}.ts`)
-        );
-        return resolved;
+        return path.resolve(`${importee}.ts`);
       }
     }
   };
 }
 
+function generateDepFile({ outputFile, sourceFiles = [], configFiles = [] }) {
+  let timestamp = Date.now();
+
+  // Save the depfile just before the node process exits.
+  process.once("beforeExit", () =>
+    writeDepFile({ outputFile, sourceFiles, configFiles, timestamp })
+  );
+
+  return {
+    name: "depfile",
+    load(sourceFile) {
+      // The 'globals' plugin adds generated files that don't exist on disk.
+      // Don't add them to the depfile.
+      if (/^[0-9a-f]{30}$/.test(sourceFile)) {
+        return;
+      }
+      sourceFiles.push(sourceFile);
+      // Remember the time stamp that we last resolved a dependency.
+      // We'll set the last modified time of the depfile to that.
+      timestamp = new Date();
+    }
+  };
+}
+
+function writeDepFile({ outputFile, sourceFiles, configFiles, timestamp }) {
+  const buildDir = process.cwd();
+  const outputDir = path.dirname(outputFile);
+
+  // Assert that the discovered bundle inputs are files that exist on disk.
+  sourceFiles.forEach(f => fs.accessSync(f));
```

Comment:
This throws if the file doesn't exist?  I feel like `assert(fs.existsSync(f))` would be more readable.

Reply:
Oh I was so entertained that I finally found a place where fs.access() had some modest utility.